packages/lit-element/src/hydrate-support.ts

```diff
@@ -30,31 +31,75 @@ interface PatchableLitElement extends HTMLElement {
 }: {
   LitElement: PatchableLitElement;
 }) => {
-  // Capture whether we need hydration or not
-  const createRenderRoot = LitElement.prototype.createRenderRoot;
-  LitElement.prototype.createRenderRoot = function (this: PatchableLitElement) {
+  const observedAttributes = Object.getOwnPropertyDescriptor(
```

Comment:
I'd like to briefly consider putting this code into `ReactiveElement` with perhaps a more generic attribute name.

If you boot up with a `shadowRoot`, we could set a `hasDeclarativeShadowRoot` property.

Perhaps we could just use `disabled` or `inert` (or prefix this with `lit-`) to do what `defer-hydration` is doing? It adds very little code and could be more generally useful. It could also be dynamic by using `shouldUpdate`?

In addition to the code, this would add a `hasAttribute` check to initialization.

Reply:
Via offline chat we decided this wasn't worth doing.

Reply:
Decided against, due to multi-actor issue; `defer-hydration` can only be used by ssr/hydration code.